std-bits/table/src/main/java/org/enso/table/data/column/operation/UnaryOperation.java

```diff
@@ -0,0 +1,204 @@
+import org.enso.table.data.column.operation.map.MapOperationProblemAggregator;
+import org.enso.table.data.column.operation.unary.*;
+import org.enso.table.data.column.storage.*;
+import org.enso.table.data.column.storage.type.StorageType;
+import org.enso.table.data.table.Column;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Value;
+
+/** A UnaryOperation is an operation that can be applied to a single ColumnStorage. */
+public interface UnaryOperation {
+  String TRUNCATE = "truncate";
+  String FLOOR = "floor";
+  String CEIL = "ceil";
+  String TEXT_LENGTH = "text_length";
+  String IS_NAN = "is_nan";
+  String IS_EMPTY = "is_empty";
+  String IS_INFINITE = "is_infinite";
+  String IS_NOTHING = "is_nothing";
+  String NOT = "not";
+  String YEAR = "year";
+  String QUARTER = "quarter";
+  String MONTH = "month";
+  String WEEK = "week";
+  String DAY = "day";
+  String HOUR = "hour";
+  String MINUTE = "minute";
+  String SECOND = "second";
+  String MILLISECOND = "millisecond";
+  String MICROSECOND = "microsecond";
+  String NANOSECOND = "nanosecond";
+
+  /** Applies the operation to the given Column. If an unsupported by the operation returns null */
+  static Column apply(
+      Column column,
+      String operationName,
+      String newColumnName,
+      MapOperationProblemAggregator problemAggregator) {
+    UnaryOperation operation = UnaryOperation.getInstance(operationName);
+
+    ColumnStorage storage = column.getStorage();
+
+    // If the storage has an inferred storage (e.g. a Mixed column) and the first level can't do get
+    // an inferred storage.
+    if (!operation.canApply(storage)
+        && storage instanceof ColumnStorageWithInferredStorage withInferredStorage) {
+      storage = withInferredStorage.getInferredStorage();
+    }
+
+    if (!operation.canApply(storage)) {
+      return null;
+    }
+
+    var result = operation.apply(column.getStorage(), problemAggregator);
+    return new Column(newColumnName, (Storage<?>) result);
+  }
+
+  static Column mapFunction(
+      Column column,
+      Function<Object, Value> function,
+      boolean nothingUnchanged,
+      StorageType expectedResultType,
+      String newColumnName,
+      MapOperationProblemAggregator problemAggregator) {
+    Builder storageBuilder =
+        Builder.getForType(expectedResultType, column.getSize(), problemAggregator);
+    applyStorageInner(
+        column.getStorage(),
+        nothingUnchanged,
+        storageBuilder,
+        i -> {
+          Value result = function.apply(column.getStorage().getItemAsObject(i));
+          Object converted = Polyglot_Utils.convertPolyglotValue(result);
+          storageBuilder.appendNoGrow(converted);
+        });
+    return new Column(newColumnName, storageBuilder.seal());
+  }
+
+  /** Gets the Operation with the given name. */
+  static UnaryOperation getInstance(String name) {
+    return switch (name) {
+      case IS_NOTHING -> IsNothingOperation.INSTANCE;
```

Comment:
Do we still need this mapping between names and classes?

It feels a bit redundant. We could just pass an instance of `UnaryOperation` in Enso code at this point.

Reply:
Good point - will remove that structure as well.